apps/hash-graph/libs/graph/src/knowledge/query.rs

```diff
@@ -134,6 +134,41 @@ pub enum EntityQueryPath<'p> {
     /// [`EntityMetadata`]: graph_types::knowledge::entity::EntityMetadata
     /// [`EntityTemporalMetadata`]: graph_types::knowledge::entity::EntityTemporalMetadata
     CreatedAtDecisionTime,
+    /// The timestamp of the transaction time when the [`Entity`] was _first undrafted_ in the
+    /// database.
+    ///
+    /// ```rust
+    /// # use serde::Deserialize;
+    /// # use serde_json::json;
+    /// # use graph::knowledge::EntityQueryPath;
+    /// let path = EntityQueryPath::deserialize(json!(["firstNonDraftCreatedAtTransactionTime"]))?;
+    /// assert_eq!(path, EntityQueryPath::FirstNonDraftCreatedAtTransactionTime);
+    /// # Ok::<(), serde_json::Error>(())
+    /// ```
+    ///
+    /// [`StructuralQuery`]: crate::subgraph::query::StructuralQuery
+    /// [`EntityMetadata`]: graph_types::knowledge::entity::EntityMetadata
+    /// [`EntityTemporalMetadata`]: graph_types::knowledge::entity::EntityTemporalMetadata
+    FirstNonDraftCreatedAtTransactionTime,
+    /// The timestamp of the decision time when the [`Entity`] was _first undrafted_ in the
+    /// database.
+    ///
+    /// This does not take into account if the [`Entity`] was updated with an earlier decision
+    /// time.
+    ///
+    /// ```rust
+    /// # use serde::Deserialize;
+    /// # use serde_json::json;
+    /// # use graph::knowledge::EntityQueryPath;
+    /// let path = EntityQueryPath::deserialize(json!(["firstNonDraftCreatedAtTransactionTime"]))?;
+    /// assert_eq!(path, EntityQueryPath::FirstNonDraftCreatedAtTransactionTime);
```

Comment:
This needs updating on `CreatedAtDecisionTime` also
```suggestion
    /// let path = EntityQueryPath::deserialize(json!(["firstNonDraftCreatedAtDecisionTime"]))?;
    /// assert_eq!(path, EntityQueryPath::FirstNonDraftCreatedAtDecisionTime);
```